### PR TITLE
feat(schema, api-client): Migrate event schemas and types to @keyshade/schema

### PR DESCRIPTION
**Pull request by muntaxir4**
## Description

**`@keyshade/schema`**
- Add schemas, types and enums for event.
- Add tests for event schemas.
- Export event schemas from `index` and its types  from `index.types` 
- Modify ProjectSchema to use accessLevelEnum
- Add simple tests for `enums.ts`.
- Change `"main": "dist/src/index.js"` to `"main": "dist/src/index.types.js"` in `package.json` due to default export being `index.types.ts`.

**`@keyshade/api-client`**
- Remove existing `event.types.d.ts`.
- Change imports for event types to `@keyshade/schema`.  <br />

Related to #519

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [x] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [x] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
enhancement, tests

___

### **Description**
- Migrated event schemas and types from `api-client` to `@keyshade/schema`.
- Added new enums for event sources, triggerers, and severities.
- Created and exported event schemas and types using zod in the `schema` package.
- Updated `ProjectSchema` to use `projectAccessLevelEnum`.
- Added tests for the new event schemas and enums.
- Updated `package.json` to reflect changes in the main entry point.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>event.ts</strong><dd><code>Update event type imports to use schema package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/api-client/src/controllers/event.ts

- Updated imports for event types to use `@keyshade/schema`.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-0e378af00f5446cbf91f6c601f547840229a8d61499c4811f589612cf478f3c0">+1/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>event.types.d.ts</strong><dd><code>Remove deprecated event types file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/api-client/src/types/event.types.d.ts

<li>Removed the entire file as event types are now in <code>@keyshade/schema</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-33a6f783cb4da4d4de5d77a96272804752f6bfdb4930f98e0d44fd4bc5c9679f">+0/-82</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>enums.ts</strong><dd><code>Add event-related enums</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/enums.ts

- Added enums for event source, triggerer, and severity.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-4a5bedad130954fdfa603e5ff11c786d350afca56f934d5bad8060b60e0776b9">+14/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Add event schemas using zod</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/event/index.ts

<li>Created schemas for <code>GetEventsRequest</code> and <code>GetEventsResponse</code>.<br> <li> Utilized new event enums in schemas.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-1b98dc2071de1bad81f6136c16d3d19d4f58c7323e1acf8dd9bea4a6475888bf">+40/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.types.ts</strong><dd><code>Define event types using zod inference</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/event/index.types.ts

<li>Defined types for <code>GetEventsRequest</code> and <code>GetEventsResponse</code> using zod <br>inference.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-aeb9dd415a499415a8371db97f01d3dbe1a4d5c2d30d6116170e4343a22e1f8a">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Export event schemas from index</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/index.ts

- Exported event schemas from the main index file.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-5c37949968eac79dd5d1c0df1176a9073fd624d2be2742d3c7d718e318aa2cdc">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.types.ts</strong><dd><code>Export event types from index types</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/index.types.ts

- Exported event types from the main index types file.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-47f389a66e4515c1526d27bc56ae56141418a0a0419271b98077614f7778d3e5">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Use project access level enum in schema</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/project/index.ts

- Updated `ProjectSchema` to use `projectAccessLevelEnum`.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-df09bc030c73369c6fdd21ddca0e63f68b1151a33247495c24a1b8564c412b67">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>enums.test.ts</strong><dd><code>Add tests for event enums</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/tests/enums.test.ts

- Added tests for event-related enums.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-7857b1442b713837779e4cb4cbc57f8a3812e028f4f233e39e5ea2f70457c910">+149/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>event.spec.ts</strong><dd><code>Add tests for event schemas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/tests/event.spec.ts

<li>Added tests for <code>GetEventsRequestSchema</code> and <code>GetEventsResponseSchema</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-03b60f9706eaef27b2a4000a4e3a1054de2d2d1456e27ca213dcd33bdbdd5555">+128/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>project.spec.ts</strong><dd><code>Update project schema tests for access level enum</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/tests/project.spec.ts

- Updated tests to reflect changes in `ProjectSchema` access level.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-4d04c509cd4ab44ec5c87dc8bcc491d52f99f8fab4499aaa08459daa168f704f">+8/-8</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Update main entry point in package.json</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/package.json

- Changed main entry point to `dist/src/index.types.js`.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/546/files#diff-ccb4b7623a0b0dfc801a63df0d5be8897e2a7fa73a9b4ada3e99661b079af7cf">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by muntaxir4** [comment]
Fixed ProjectSchema issues. This could be used in #540

File: packages/schema/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@keyshade/schema",
   "version": "1.0.0",
-  "main": "dist/src/index.js",
+  "main": "dist/src/index.types.js",
```
Comment (rajdip-b) [inline packages/schema/package.json:4]: I'm not sure if this change would be a good thing to make. What are your thoughts?

Comment (muntaxir4) [reply packages/schema/package.json:4]: Since '.' import would be 'index.types.js'. I was able to import it in packages easily without moduleResolution.

Comment (muntaxir4) [reply packages/schema/package.json:4]: Should I include this or keep it just index.js? @rajdip-b

Comment (rajdip-b) [reply packages/schema/package.json:4]: Ah! okay then that works for me.

**Review by rajdip-b — APPROVED** [review]

**Pull request merged**